components/ApiKeyInput.tsx

```diff
@@ -1,27 +1,26 @@
-import { useState, useEffect } from 'react';
+import { useMemo } from 'react';
 
-export default function Component({defaultKey = null, keyName}) {
-    const [apiKey, setApiKey] = useState(() => {
-        const storedKey = window.localStorage.getItem(keyName);
-        return storedKey || defaultKey || '';
-    });
+interface Props {
+    keyName: string;
+    defaultKey?: string;
+    onChange?: (key: string) => void;
+}
```

Comment:
Added this definition just to make coding easier; TypeScript can now verify the correct parameters are being passed in to the function.